swatch-contracts/src/main/java/com/redhat/swatch/contract/security/RhIdentityHeaderAuthenticationMechanism.java

```diff
@@ -60,7 +60,7 @@ public Uni<SecurityIdentity> authenticate(
       return Uni.createFrom().nullItem();
     }
     try {
-      RhIdentityPrincipal identity = RhIdentityPrincipal.fromHeader(xRhIdentityHeader);
+      RhIdentityPrincipal identity = new RhIdentityPrincipal.Factory().fromHeader(xRhIdentityHeader);
```

Comment:
I see that you inject ObjectMapper into the factory... This causes NPEs in a couple of use patterns. I personally don't mind either constructing per-instance or having a static ObjectMapper manually instantiated for those use cases instead.

Reply:
Yeah, I fixed that.  The code can now just `Inject` the RhIdentityPrincipalFactory to get a configured instance and then build the Principal from it